packages/iTwinUI-react/src/core/Dialog/DialogDragContext.tsx

```diff
@@ -0,0 +1,20 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+
+export type DialogDragContextProps = {
+  onPointerDown: (e: React.PointerEvent<HTMLElement>) => void;
+};
+
+export const DialogDragContext = React.createContext<
+  DialogDragContextProps | undefined
+>(undefined);
+
+export const useDialogDragContext = () => {
+  const context = React.useContext(DialogDragContext);
```

Comment:
We have this `useSafeContext` hook, if you wanna reuse...

Reply:
It does a different thing. My hook doesn't throw it if component is not under context e.g. custom dialog body.